Public/Src/Engine/Processes/ReportedFileAccess.cs

```diff
@@ -622,6 +623,69 @@ public bool TryParseAbsolutePath(PipExecutionContext context, Process pip, out A
                 enumeratePattern);
         }
 
+        /// <nodoc />
+        public void Serialize(
+            BuildXLWriter writer,
+            Dictionary<ReportedProcess, int> processMap,
+            Action<BuildXLWriter, AbsolutePath> writePath)
+        {
+            writer.Write((byte)Operation);
+
+            if (processMap != null && processMap.TryGetValue(Process, out int index))
+            {
+                writer.WriteCompact(index);
+            }
+            else
+            {
+                Process.Serialize(writer);
+            }
+
+            writer.WriteCompact((int)RequestedAccess);
+            writer.WriteCompact((int)Status);
+            writer.Write(ExplicitlyReported);
+            writer.Write(Error);
+            writer.Write(Usn.Value);
+            writer.Write((uint)DesiredAccess);
+            writer.Write((uint)ShareMode);
+            writer.Write((uint)CreationDisposition);
+            writer.Write((uint)FlagsAndAttributes);
+
+            if (writePath != null)
+            {
+                writePath.Invoke(writer, ManifestPath);

```

Comment:
>Invoke [](start = 26, length = 6)

Why are you using `.Invoke` here? Please just use `writePath(writer, ManifestPath)`

Reply:
I made it optional at first, so I was writing something like writePath?.Invoke(..). But then I changed it, and forget to remove the invoke.

---
In reply to: [276817065](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276817065) [](ancestors = 276817065)